src/extensions/mountainview/MVSortingView/VisibleElectrodesControl.tsx

```diff
@@ -0,0 +1,49 @@
+import { Button, Checkbox, Radio } from '@material-ui/core';
+import React, { FunctionComponent, useCallback, useEffect, useMemo, useState } from 'react';
+import { RecordingInfo, Sorting, SortingCuration, SortingInfo, SortingSelection, SortingSelectionDispatch } from '../../extensionInterface';
+
+type Props = {
+    recordingInfo: RecordingInfo
+    selection: SortingSelection
+    selectionDispatch: SortingSelectionDispatch
+}
+
+const VisibleElectrodesControl: FunctionComponent<Props> = ({ recordingInfo, selection, selectionDispatch }) => {
+    const [mode, setMode] = useState<'all' | 'custom'>('all')
+    const [customElectrodeIds, setCustomElectrodeIds] = useState<number[]>([])
+    const { selectedElectrodeIds } = selection
+
+    const handleModeAll = useCallback(() => {
+        setMode('all')
+    }, [setMode])
+    const handleModeCustom = useCallback(() => {
+        setMode('custom')
+    }, [setMode])
+    useEffect(() => {
+        const includeElectrode = (eid: number) => {
+            switch(mode) {
+                case 'all': return true
+                case 'custom': return customElectrodeIds.includes(eid)
+                default: throw Error('Unexpected mode')
+            }
+        }
+        const visibleElectrodeIds = (recordingInfo?.channel_ids || []).filter(eid => includeElectrode(eid))
+        selectionDispatch({type: 'SetVisibleElectrodeIds', visibleElectrodeIds})
+    }, [recordingInfo, selectionDispatch, mode, customElectrodeIds])
+    const handleRestrictToSelected = useCallback(() => {
+        setMode('custom')
+        setCustomElectrodeIds(selectedElectrodeIds || [])
+    }, [selectedElectrodeIds, setMode, setCustomElectrodeIds])
+
+    const hasSelection = useMemo(() => ((selectedElectrodeIds || []).length > 0), [selectedElectrodeIds])
+
+    return (
+        <div>
+            <span style={{whiteSpace: 'nowrap'}}><Radio checked={mode === 'all'} onClick={handleModeAll}/> Show all</span>
+            <span style={{whiteSpace: 'nowrap'}}><Radio checked={mode === 'custom'} onClick={handleModeCustom} /> Show custom</span>
+            <Button disabled={!hasSelection} onClick={handleRestrictToSelected}>Restrict to selected</Button>
+        </div>
```

Comment:
So the intended user interaction is "I am showing all, I select some, then I click 'restrict to selected' and the radio button toggles to 'custom' and the view is restricted; and when I want to go back, I'll click the 'show all' radio button again," right?
Do you lose your custom view if you go back to 'show all'?

Reply:
no it remembers, you can switch back and forth